lib/stats/series.js

```diff
@@ -0,0 +1,209 @@
+import base from 'taskcluster-base';
+
+const BASE_WORKER_SCHEMA = {
+    workerId: base.stats.types.String,
+    workerType: base.stats.types.String,
+    workerGroup: base.stats.types.String,
+    provisionerId: base.stats.types.String
+};
+
+export const workerStart = new base.stats.Series({
+  name: 'worker.start',
+  columns: Object.assign({}, BASE_WORKER_SCHEMA, {
+    // Time worker started
+    value: base.stats.types.Number
+  })
+});
+
+export const workerShutdown = new base.stats.Series({
+  name: 'worker.shutdown',
+  columns: Object.assign({}, BASE_WORKER_SCHEMA, {
+    // Time when worker received shutdown signal
+    value: base.stats.types.Number
+  })
+});
+
+export const workerUptime = new base.stats.Series({
+  name: 'worker.uptime',
+  columns: Object.assign({}, BASE_WORKER_SCHEMA, {
+    // Time when worker received shutdown signal
+    value: base.stats.types.Number
+  })
+});
+
+
+export const workerTotalMemory = new base.stats.Series({
+  name: 'worker.memory_total',
+  columns: Object.assign({}, BASE_WORKER_SCHEMA, {
+    // Time when worker received shutdown signal
+    value: base.stats.types.Number
+  })
+});
+
+export const workerFreeMemory = new base.stats.Series({
+  name: 'worker.memory_free',
+  columns: Object.assign({}, BASE_WORKER_SCHEMA, {
+    // Time when worker received shutdown signal
+    value: base.stats.types.Number
+  })
+});
+
+export const workerHDUsed = new base.stats.Series({
+  name: 'worker.hd_used',
+  columns: Object.assign({}, BASE_WORKER_SCHEMA, {
+    // Time when worker received shutdown signal
+    value: base.stats.types.Number
+  })
+});
+
+export const workerHDFree = new base.stats.Series({
+  name: 'worker.hd_free',
+  columns: Object.assign({}, BASE_WORKER_SCHEMA, {
+    // Time when worker received shutdown signal
+    value: base.stats.types.Number
+  })
+});
+
+export const workerHDTotal = new base.stats.Series({
+  name: 'worker.hd_total',
+  columns: Object.assign({}, BASE_WORKER_SCHEMA, {
+    // Time when worker received shutdown signal
+    value: base.stats.types.Number
+  })
+});
```

Comment:
I was tempted to do that, but then decided to follow the crowd and store one value per series :).

Can we not do queries across series?